service/history/historyEventNotifier.go

```diff
@@ -177,7 +177,7 @@ func (notifier *historyEventNotifierImpl) dispatchHistoryEventNotification(event
 
 func (notifier *historyEventNotifierImpl) enqueueHistoryEventNotification(event *historyEventNotification) {
 	// set the timestamp just before enqueuing the event
-	event.timestamp = time.Now()
+	event.timestamp = common.NewRealTimeSource().Now()
```

Comment:
Can you have time source as a field on historyEventNotifierImpl so another implementation could be injected.